qiskit/synthesis/evolution/product_formula.py

```diff
@@ -165,9 +165,9 @@ def _two_qubit_evolution(pauli, time, cx_structure):
     elif all(labels == "Z"):  # RZZ
         definition.rzz(2 * time, qubits[0], qubits[1])
     elif labels[0] == "Z" and labels[1] == "X":  # RZX
-        definition.rzx(2 * time, qubits[1], qubits[0])
-    elif labels[0] == "X" and labels[1] == "Z":  # RXZ
         definition.rzx(2 * time, qubits[0], qubits[1])
+    elif labels[0] == "X" and labels[1] == "Z":  # RXZ
+        definition.rzx(2 * time, qubits[1], qubits[0])
```

Comment:
Do we have explicit tests of these two cases somewhere?  I feel like yes because I remember some PR because it was the "wrong" way round before, but just to check.

Reply:
We had one for XZ, I added another one for the other way around.